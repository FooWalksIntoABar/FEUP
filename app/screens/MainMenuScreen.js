import React from 'react'
import { View, TouchableOpacity, Image, Text, ScrollView } from 'react-native'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Images from '../assets/images/images'
import styles from '../styles/MainMenu.style'
import { registerForPushNotificationsAsync } from '../helpers/Notification'
import EnvVars from '../constants/EnviromentVars'
import { addChild } from '../actions/childActions'

export class MainMenuScreen extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      kids: []
    }
  }

  componentDidMount () {
    registerForPushNotificationsAsync()
    this.getChildren()
  }

  getChildren () {
    let url = `${EnvVars.apiUrl}routine_manager/children?userEmail=${this.props.loggedUserEmail}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.setState({ kids: JSON.parse(responseJson.response)
          })
        } else {
          console.log('Ta mal')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  onChildClick = (i) => {
      console.log(this.state.kids[i])
      this.props.addChild(this.state.kids[i])
      this.props.navigation.navigate('ChildMainMenu')
  }

  render () {
    const childButtons = this.state.kids.map((kid, index) => (
      <View style={styles.childContainer} key={index}>
        <TouchableOpacity
          class='child'
          style={styles.button}
          activeOpacity={0.8}
          onPress={() => this.onChildClick(index)}>
          <Image style={styles.buttonImage} source={{ uri: kid.image }} resizeMode={'cover'} />
        </TouchableOpacity>
        <Text style={styles.childName}>{kid.name}</Text>
      </View>))
    return (
      <View style={styles.mainMenuContainer}>
        <TouchableOpacity style={styles.infoButton} onPress={() => this.props.navigation.navigate('AppIntro')}>
          <Image style={styles.buttonImage} source={Images.ui.info} resizeMode={'contain'} />
        </TouchableOpacity>
        <ScrollView style={styles.childButtonsContainer}>
          {childButtons}
        </ScrollView>
        <View style={styles.parentContainer}>
          <TouchableOpacity
            style={styles.parentButton}
            activeOpacity={0.8}
            onPress={() => this.props.navigation.navigate('ParentMainMenu')}>
            <Text style={styles.parentTooltip}>Zona de pai</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    loggedUserEmail: state.user.email
  }),
    /* istanbul ignore next */
  dispatch => ({
    addChild: child => dispatch(addChild(child))
  })

)(MainMenuScreen)

MainMenuScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  loggedUserEmail: PropTypes.string.isRequired,
  addChild: PropTypes.func.isRequired
}
