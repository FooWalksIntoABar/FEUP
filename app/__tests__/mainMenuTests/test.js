import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import MainMenuScreen from '../../screens/MainMenuScreen'

describe('MainMenuScreen snapshot', () => {
  it('renders MainMenuScreen correctly', async () => {
    const tree = renderer.create(<MainMenuScreen
      navigation={{}} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
