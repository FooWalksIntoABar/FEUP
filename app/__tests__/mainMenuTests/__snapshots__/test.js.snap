// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuScreen snapshot renders MainMenuScreen correctly 1`] = `
<Component
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#3db3e2",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <Styled(Spinner)
      color="#FFA23D"
    />
  </ScrollViewMock>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#046488",
          "borderTopLeftRadius": 45,
          "borderTopRightRadius": 45,
          "elevation": 10,
          "height": 150,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 450,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "LinotteBold",
            "fontSize": 30,
            "marginTop": 15,
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 3,
          }
        }
      >
        Zona de pai
      </Text>
    </TouchableOpacity>
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Object {
        "height": 30,
        "position": "absolute",
        "right": 20,
        "top": 20,
        "width": 30,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </TouchableOpacity>
</Component>
`;
